test/common.js

```diff
@@ -265,3 +265,17 @@ export function getBody(win? : SameDomainWindowType = window) : HTMLBodyElement
 
     return win.document.body;
 }
+
+// eslint-disable-next-line no-restricted-globals, promise/no-native
+export async function loadScript(url : string) : Promise<void> {
+    const scriptelement = document.createElement('script');
+    scriptelement.setAttribute('src', url);
+    getBody().prepend(scriptelement);
+
+    // eslint-disable-next-line no-restricted-globals, promise/no-native
+    const scriptPromise = new Promise((resolve) => {
+        scriptelement.addEventListener('load', () => resolve());
+    });
+
+    await scriptPromise;
```

Comment:
Why does `scriptPromise` need to be a Promise?  We will only resolve on load of script.  Also, what about loading error?

Reply:
In our tests, we want to be sure that the script is loaded.
For example, when dynamically loading the vue script, we want to be sure that the `window.Vue` is defined before creating the zoid component.

The `<script>` tags on html usually blocks the javascript executions until the file is loaded. But if it is added at runtime, as we are doing here, the document will not interpret it right away and `window.Vue` will not be available until we await for the `onload` event of this `<script>` tag.

Regarding loading errors, the `loadScript` function is used to load scripts that are served locally by karma itself. Even if it was possible to fail, we already have a native error like:
`GET file:///base/test/lib/vue_v2.5.16.runtime.min.js::ERR_FILE_NOT_FOUND`
But if you want to add a **onerror** event anyway, I will be happy to add it.

Reply:
@andregardi I see what you're doing.  I typically would just return a new Promise and resolve on load of the script rather than initialize a Promise and await on it.  It's in the tests so no biggy.

Reply:
@mnicpt today I read that again with fresh eyes and it makes complete sense to return the promise instead await it.
I changed that, thank you for pointing that out.